### DISCUSSION TITLE
How to validate coordinates are in bounds?

### DISCUSSION THREAD

**Discussion by mkoskl** [dvdoug/PHPCoord#67] [Q&A] [answered]
Hello! 

How can I validate coordinates are in bounds of specified CRS?

E.g. if I have URN for the coordinate system and latitude / longitude pair, how can I check if the latitude and longitude are valid for specified CRS?

Br, Miika

**Comment by dvdoug** [comment, marked as answer]
Interesting, that's not a use case I've ever considered - I've always assumed that you've got the lat/long it must be valid 😱 

It depends how accurate you need the check to be. The bounding areas for all CRS types (even non-geographic ones) are defined in WGS84 coordinates so the most accurate way that works for all point types would be this (slightly clunky API):

```php
$crs = Geographic2D::fromSRID(Geographic2D::EPSG_ED50);
$point = GeographicPoint::create( // New York, will fail
    latitude: new Degree(40.689167),
    longitude: new Degree(-74.044444),
    crs: $crs
);
$wgs84 = Geographic2D::fromSRID(Geographic2D::EPSG_WGS_84);
$isInsideCRS = $crs->getBoundingArea()->containsPoint($point->convert($wgs84)->asGeographicValue());
```

**Reply by mkoskl** [reply]
Hi, and thanks for answer! Our usecase is a cultural environment (buildings etc.) management software, where user can input coordinates manually in various CRS. Of course manual input is prone to typing errors or user could input correct coordinates but choose invaldi CRS etc. and we would like to prevent those, at least some of them.

**Reply by dvdoug** [reply]
👍 I'll add a nicer API for this

**Reply by mkoskl** [reply]
@dvdoug I see you added this on 5.10.0. Thanks!
